### ISSUE TITLE
Ansi sequences stripped?

### ISSUE THREAD

**Issue by bend-n** [ratatui-org/ratatui#563]
## Description
Im not allowed to add my own ANSI escape sequences.

## To Reproduce
Change the `hello_world.rs` example's `render_app` function to
```rs
fn render_app(frame: &mut Frame) {
    let greeting = Paragraph::new("\x1b[0;34;31mHello World!\x1b[0m (press 'q' to quit)");
    frame.render_widget(greeting, frame.size());
}
```
This occurs:

![image](https://github.com/ratatui-org/ratatui/assets/70787919/6948f447-3fd3-4e65-b307-52671984be02)

Piping the output reveals that the output is
```
[0;34;31mHello[1;16HWorld![0m[1;26H(press[1;33H'q'[1;37Hto[1;40Hquit)
```

## Expected behavior

Red `Hello World!`.

![image](https://github.com/ratatui-org/ratatui/assets/70787919/af927ed9-2562-4c77-acf5-0c17571d1dd6)

## Environment
- OS: Linux
- Terminal Emulator: kitty
- Font: Cascadia code
- Crate version: [`main`](https://github.com/ratatui-org/ratatui/commit/089f8ba66a50847780c4416b9b8833778a95e558)
- Backend: crossterm

## Additional context

I have reason to believe this is a feature; but im not sure how to turn it off.

Can you direct me to the part of the code that strips my `\x1b`, so i can remove it?

**Comment by kdheepak** [comment]
Ratatui doesn't work that way unfortunately. 

The smallest unit of data in Ratatui is a `Cell`:

https://github.com/ratatui-org/ratatui/blob/e5caf170c8c304b952cbff7499fd4da17ab154ea/src/buffer.rs#L15-L26

You can see that a `Cell` has a `String` which is the symbol that will be drawn in that `Cell` as well as `Style` information (foreground color, background color, modifiers etc).

And a `Buffer` is a collection of `Cell`s:

https://github.com/ratatui-org/ratatui/blob/e5caf170c8c304b952cbff7499fd4da17ab154ea/src/buffer.rs#L149-L157

`Buffer` represents the `Area` in which the `Cell`s are going to be drawn.

When you render an instance of `Paragraph`, the `Paragraph` widget is passed to a `render` function along with a `Buffer`.

https://github.com/ratatui-org/ratatui/blob/e5caf170c8c304b952cbff7499fd4da17ab154ea/src/widgets/paragraph.rs#L253-L268

The implementation of `render` loops through the content of the string in the `Paragraph` widget and puts the content in appropriate `Cell`s using methods exposed on the `Buffer` struct.

Basically, any styling has to happen though the `Style` struct (i.e. methods that apply `Style`s to `Cell`s) and you can't use ansi escape sequences directly in the string content.

As a workaround, what you can do however is convert the ansi escape sequences into a `Style` struct. There's a crate for this: https://crates.io/crates/ansi-to-tui

**Comment by bend-n** [comment]
Hmm, thank you.

I guess it makes sense that the screen is represented as a `[char; N * M]`.

For some background, i want to do this because i made [a cool crate](https://docs.rs/comat) for ansi coloring, so i was a little disappointed i couldnt use it with ratatui, i suppose making a little macro that appends `into_text` isnt that hard.

**Comment by kdheepak** [comment]
You may be interested in this discussion: https://github.com/ratatui-org/ratatui/issues/493

We are interested in simplifying applying styles with a macro. Contributions to that end would be very welcome. If you want to chime in on that thread with your thoughts that'd be great.
